.github/workflows/finalize_release.yml

```diff
@@ -0,0 +1,65 @@
+name: Finalize Release
+
+# Every time we trigger the workflow on a branch.
+on:
+  workflow_dispatch:
+
+jobs:
+  generate_version_name:
+    runs-on: ubuntu-latest
+
+    outputs:
+      version_name: ${{ steps.version_name.outputs.PROJECT_VERSION }}
+
+    steps:
+      - uses: actions/checkout@v4
+
+      # Get the version name from a script and save it in version_name output.
+      - name: Generate version_name
+        id: version_name
+        run: |
+          echo "▸ Set run permission."
+          chmod +x scripts/version_name.sh
+          echo "▸ Getting version name"
+          PROJECT_VERSION=$(./scripts/version_name.sh)
+          echo "▸ Creating the version_name output with value: $PROJECT_VERSION"
+          echo "PROJECT_VERSION=$PROJECT_VERSION" >> $GITHUB_OUTPUT
+          echo "▸ DONE"
+
+  publish_draft_release:
+    runs-on: ubuntu-latest
+    needs: generate_version_name
+
+    steps:
+      - uses: actions/checkout@v4
+
+      # Publish Github release and the tag
+      - name: Publish Github release
+        uses: ncipollo/release-action@v1
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          VERSION_NAME: ${{ needs.generate_version_name.outputs.version_name }}
+        with:
+          token: ${{ env.GITHUB_TOKEN }}
+          tag: ${{ env.VERSION_NAME }}
+          allowUpdates: true
+          updateOnlyUnreleased: true
+          omitNameDuringUpdate: true
+          omitBodyDuringUpdate: true
+          draft: false
+
+  remove_release_branch:
+    runs-on: ubuntu-latest
+    needs: publish_draft_release
+
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Remove release branch
+        uses: dawidd6/action-delete-branch@v3
```

Comment:
Not sure what does action all does, but wouldn't it be easier to use plain git?

Reply:
This action allows deleting multiple branches and has some filtering options, but in our case is indeed not necessary.